packages/tools/src/counter.ts

```diff
@@ -0,0 +1,20 @@
+import throttle from 'lodash.throttle';
+
+export function throttledCounter(fn: (counter: number) => unknown, wait: number) {
+	let counter = 0;
+
+	const throttledFn = throttle(
+		() => {
+			fn(counter);
+
+			counter = 0;
+		},
+		wait,
+		{ leading: false },
+	);
+
+	return () => {
+		counter++;
+		throttledFn();
+	};
+}
```

Comment:
I dont think this should be placed here, its not quite useful outside that scope

not sure about the `lodash` necessity as well